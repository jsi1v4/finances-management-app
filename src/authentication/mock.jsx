import React from 'react';

import { AuthenticationContext } from './index';

export function AuthenticationMock({ children }) {
  return (
    <AuthenticationContext.Provider
      value={{
        isAuth: true,
        userName: 'John Doe',
        authOn: jest.fn(),
        authOff: jest.fn()
      }}
    >
      {children}
    </AuthenticationContext.Provider>
  );
}

export default AuthenticationMock;
