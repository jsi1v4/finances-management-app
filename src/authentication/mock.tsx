import React from 'react';

import { AuthenticationContext } from './index';

export function AuthenticationMock({ children }) {
  return (
    <AuthenticationContext.Provider
      value={{
        isAuth: true,
        session: null,
        authOn: jest.fn(),
        authOff: jest.fn()
      }}
    >
      {children}
    </AuthenticationContext.Provider>
  );
}

export default AuthenticationMock;
