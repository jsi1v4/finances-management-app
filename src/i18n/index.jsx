import React, { createContext, useState } from 'react';
import { useIntl } from 'react-intl';

import { INIT_LOCALE } from '../config';

export function useI18n() {
  const intl = useIntl();
  return (id, values) => {
    const message = intl.formatMessage({ id }, values);
    return <>{message}</>;
  };
}

export const I18nContext = createContext(null);

export function I18nProvider({ children }) {
  const [lang, setLanguage] = useState(INIT_LOCALE);
  const changeLang = (value) => {
    setLanguage(value);
  };

  return (
    <I18nContext.Provider
      value={{
        lang,
        changeLang
      }}
    >
      {children}
    </I18nContext.Provider>
  );
}

export default I18nProvider;
