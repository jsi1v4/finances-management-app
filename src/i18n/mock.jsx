import React from 'react';

import { I18nProvider } from './index';
import { I18nMessagesProvider } from './messages';

export function I18nMock({ children }) {
  return (
    <I18nProvider>
      <I18nMessagesProvider>{children}</I18nMessagesProvider>
    </I18nProvider>
  );
}

export default I18nMock;
