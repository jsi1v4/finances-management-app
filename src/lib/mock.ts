export class Auth {
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  onAuthStateChanged(fn = () => {}) {
    return () => {};
  }

  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  async setPersistence(persist?: string) {
    return new Promise((resolve) => resolve({}));
  }

  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  async signInWithEmailAndPassword(username: string, password: string) {
    return new Promise((resolve) => resolve({}));
  }

  async signOut() {
    return new Promise((resolve) => resolve({}));
  }
}

export default Auth;
