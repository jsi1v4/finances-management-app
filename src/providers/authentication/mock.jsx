import React from 'react';

import { AuthenticationContext } from './index';

export function AuthenticationMock({ children }) {
  return (
    <AuthenticationContext.Provider
      value={{
        userName: 'John Doe',
        signOut: jest.fn(),
      }}
    >
      {children}
    </AuthenticationContext.Provider>
  );
}

export default AuthenticationMock;
