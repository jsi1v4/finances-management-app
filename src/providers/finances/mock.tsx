import React from 'react';

import { FinancesContext } from './index';
import { FinancesService } from './mock-service';

export function FinancesMock({ children }) {
  return (
    <FinancesContext.Provider value={new FinancesService({})}>
      {children}
    </FinancesContext.Provider>
  );
}

export default FinancesMock;
