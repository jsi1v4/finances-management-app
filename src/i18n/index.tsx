import React, { createContext, useContext, useState } from 'react';
import { useIntl } from 'react-intl';

import { INIT_LOCALE } from 'src/config';
import { II18nContext, Locale } from './types';

export const I18nContext = createContext<II18nContext>(null as never);

export const useI18n = () => useContext(I18nContext);

export function I18nProvider({ children }) {
  const [lang, setLanguage] = useState<Locale>(INIT_LOCALE);
  const changeLang = (value: Locale) => {
    setLanguage(value);
  };

  return (
    <I18nContext.Provider
      value={{
        lang,
        changeLang
      }}
    >
      {children}
    </I18nContext.Provider>
  );
}

export function useI18nMessage() {
  const intl = useIntl();
  return (id: string, values?: Record<string, string>) => {
    const message = intl.formatMessage({ id }, values);
    return message;
  };
}
