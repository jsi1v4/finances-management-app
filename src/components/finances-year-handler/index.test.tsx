import React from 'react';
import { render } from '@testing-library/react';

import { I18nMock } from 'src/i18n/mock';
import { FinancesYearHandler } from './index';

describe('FinancesYearHandler component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <FinancesYearHandler value={2021} onChange={jest.fn()} />,
      {
        wrapper: I18nMock
      }
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
