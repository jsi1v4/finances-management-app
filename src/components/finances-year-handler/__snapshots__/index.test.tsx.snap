// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinancesYearHandler component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-space ant-space-horizontal ant-space-align-center"
  >
    <div
      class="ant-space-item"
      style="margin-right: 8px;"
    >
      <button
        class="ant-btn ant-btn-link ant-btn-icon-only"
        type="button"
      >
        <span
          aria-label="left"
          class="anticon anticon-left"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="left"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
            />
          </svg>
        </span>
      </button>
    </div>
    <div
      class="ant-space-item"
      style="margin-right: 8px;"
    >
      <span
        class="styles__ColorText-hcptl5-0 gdVwOx"
      >
        2021
      </span>
    </div>
    <div
      class="ant-space-item"
    >
      <button
        class="ant-btn ant-btn-link ant-btn-icon-only"
        type="button"
      >
        <span
          aria-label="right"
          class="anticon anticon-right"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="right"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
